### PR TITLE
Fix for IOException while editing config and adds some Extensions for Key/Value mapped pair support

### PR DESCRIPTION
**Pull request by sohjsolwin**
Fixes an IOException caused when editing a watched config via Notepad while the application is running.
  Notepad seems to save the configuration file twice when it saves. First saving the file contents, then saving the file metadata. This occurs in a fraction of a second, but when it happens, the file watcher is already attempting to read the file while Notepad is still attempting the second save, and an IOException occurs. This PR introduces a brief delay in the event that an IOException occurs and then attempts loading the config again. If it fails a second time, it logs the failure and continues. 

Th e JsonConfigExtensions class is a small set of extensions for working with configurations in the format of:

{
  "configData": [ 
    { "Key": "data", "Value": 5 },
    { "Key": "fruit", "Value": "Bananna" }
  ]
}

It supports any inherently convertible types (integers, longs, booleans, other number types, strings, etc) and explicitly supports integer arrays.

### CODE REVIEW

File: JsonConfig/Config.cs [hunk 1]
```diff
@@ -119,18 +119,42 @@ static Config ()
 		private static FileSystemWatcher userConfigWatcher;
 		public static void WatchUserConfig (FileInfo info)
 		{
+            var lastRead = File.GetLastWriteTime(info.FullName);
```
Comment (Dynalon) [inline JsonConfig/Config.cs]: indentation is broken with your changes, could you re-indent and resubmit the PR again?

Comment (sohjsolwin) [inline JsonConfig/Config.cs]: ah, I see that. Yup, will correct.

**Comment by sohjsolwin** [comment]
Ok, I updated the config.cs to include the mentioned tab changes.

**Comment by Dynalon** [comment]
Thanks for you your contribution and that quick indentation fix!

**Comment by sohjsolwin** [comment]
Thank you for all of the hard work. I've definitely found it useful. I also just noticed that the nuget package is nearly a year our of date. Do you manage that or does raistlinthewiz handle it?

**Comment by Dynalon** [comment]
I don't know much about creating NuGet packages, I only use them. I think I got the rights to add you as a packager if you want to provide new NuGet versions.

**Comment by sohjsolwin** [comment]
Sure. I can do that. Thanks. I'll build it from the latest in master and push up the new one. I've started shifting over to using nuget for references where possible. It's surprisingly useful, but also has it's own small set of headaches.

**Pull request merged**